### PR TITLE
Create new function to import GDS file in HFSS

### LINKED ISSUE #3992: Import GDS with PyAEDT
### Description of the feature

Currently GDS import is supported with PyEDB but we got request to have it also in PyAEDT

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by amichel0205**
Create new function to import GDS file in HFSS.
It seems this function works only in Graphical Mode.

Close #3992

### CODE REVIEW

File: _unittest/test_20_HFSS.py [hunk 1]
```diff
@@ -1624,3 +1624,30 @@ def test_67_transient_composite(self, add_app):
...
+        # list of gds number
+        gds_number = [7, 9]
+        # list of elevation
+        elevation = [100e-6, 110e-6]
+        # list of thickness
+        thickness = [10e-6, 5e-6]
+        # import method 0 script, 1 Parasolid
+        importmethod = 0
+        assert self.aedtapp.import_gds_3d(gds_file, gds_number, elevation, thickness, importmethod)
+        aedtapp.close_project(save_project=False)
```
Comment (gmalinve) [inline _unittest/test_20_HFSS.py:1641]: @amichel0205 aedtapp is not an aedt instance. Why are you trying to close a project in a non existing instance?

File: _unittest/test_20_HFSS.py [hunk 2]
```diff
@@ -1624,3 +1624,30 @@ def test_67_transient_composite(self, add_app):
         aedtapp.solution_type = "Transient Composite"
         assert aedtapp.solution_type == "Transient Composite"
         aedtapp.close_project(save_project=False)
+
+    @pytest.mark.skipif(config["NonGraphical"], reason="Test fails on build machine")
+    def test_68_import_gds_3d(self):
+        self.aedtapp.insert_design("gds_import_H3D")
+        gds_file = os.path.join(local_path, "../_unittest/example_models", "cad", "GDS", "gds1.gds")
+        # list of gds number
+        gds_number = [7, 9]
```
Comment (gmalinve) [inline _unittest/test_20_HFSS.py:1633]: @amichel0205 it's a minor comment but for me you could directly pass these inputs (as well as importmethod) directly to the method without storing them in a vr.

Comment (amichel0205) [reply _unittest/test_20_HFSS.py:1633]: in reality the list can be much longer like this:
gds_number = [3, 6, 7, 8, 9, 10, 11, 30, 31, 32, 33, 34, 35]
elevation = [375.355, 375.575, 376.055, 376.655, 377.455, 378.055, 378.855, 379.455, 380.255, 380.855, 381.655, 
             382.405, 383.205]
thickness = [0.2, 0.5, 0.6, 0.8, 0.6, 0.8, 0.6, 0.8, 0.6, 0.8, 0.75, 0.8, 3]

File: _unittest/test_20_HFSS.py (hunk 1 above)
Comment (amichel0205) [reply _unittest/test_20_HFSS.py]: I will remove this line

File: _unittest/test_20_HFSS.py (hunk 2 above)
Comment (gmalinve) [reply _unittest/test_20_HFSS.py:1633]: @amichel0205 in the test I meant, you can simply pass the list without storing in variables.
It's better to do it in this way:
assert self.aedtapp.import_gds_3d(gds_file_path=gds_file, gds_number_list=gds_number [...])
so you specify the input name.

File: pyaedt/application/Analysis3D.py [hunk 3]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
         self.oeditor.ImportDXF(vArg1)
         return True
 
+    @pyaedt_function_handler
+    def import_gds_3d(self, gds_file_path, gds_number_list, elevation_list=None, thickness_list=None, import_method=1):
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1281]: @amichel0205 for me it's better to have gds_numbers rather than gds_number_list, elevation rather than elevation_list and thickness rather than thickness_list.
There's no need to specify list in the name because you are already explain it in the docstring below.

File: pyaedt/application/Analysis3D.py [hunk 4]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+        """Import a GDSII file.
+
+        Parameters
+        ----------
+        gds_file_path : str
+            Path to the GDS file.
+        gds_number_list : list
+            List of layer number to import.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1289]: @amichel0205 numbers, plural.

File: pyaedt/application/Analysis3D.py [hunk 5]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+        elevation_list : list, optional
+            List of elevation for each GDS layer number imported.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1291]: @amichel0205 I'd rephrase this sentance, it's not very clear.

File: _unittest/test_20_HFSS.py (hunk 2 above)
Comment (amichel0205) [reply _unittest/test_20_HFSS.py:1633]: It seems 
@codecov codecov/patch Failing after 1s — 7.40% of diff hit (target 85.00%)

is failing after 1s do you have any idea why

File: pyaedt/application/Analysis3D.py [hunk 6]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+            Number of element must be the same as gds_number list.
+            If not or None elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1293]: @amichel0205 The default is ````None```` in which case the thickness won't be applied.

File: pyaedt/application/Analysis3D.py [hunk 7]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+        thickness_list : list, optional
+            List of thickness for each GDS layer number imported.
+            Number of element must be the same as gds_number list.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1296]: @amichel0205 here the same, I'd rephrase the first sentance and implement the check.
The default is ``None`` in which case the thickness won't be applied. Use `` for the default values, look at the other docstrings

File: pyaedt/application/Analysis3D.py [hunk 8]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+            return False
+
+        if (
+            elevation_list is None
+            or thickness_list is None
+            or (len(gds_number_list) != len(elevation_list) or len(gds_number_list) != len(thickness_list))
+        ):
+            self.logger.warning(
+                "One of parameter list: elevation and/or thickness is either: "
+                "not specified or has different number of element than GDS number."
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1322]: @amichel0205 elements

File: pyaedt/application/Analysis3D.py [hunk 9]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+                        i,
+                        "Thickness:=",
+                        thickness_list[j],
+                        "Elevation:=",
+                        elevation_list[j],
+                        "Color:=",
+                        "color",
+                    ],
+                ]
+                ordermap = ordermap + ordermap1
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1372]: @amichel0205 use extend to append a list to another list, is nicer.

File: pyaedt/application/Analysis3D.py [hunk 10]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+                j = j + 1
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1373]: @amichel0205 if you have to work with indeces, why don't you use a for loop as:
for(j=0; j<(something);j++) ?

**Comment by gmalinve** [comment]
@amichel0205 there are still many lines not covered by test, I'd improve that.

File: pyaedt/application/Analysis3D.py (hunk 3 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1281]: I've implemented it. I run test before to commit and push

File: pyaedt/application/Analysis3D.py (hunk 4 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1289]: done

File: pyaedt/application/Analysis3D.py (hunk 3 above)
Comment (gmalinve) [reply pyaedt/application/Analysis3D.py:1281]: @amichel0205 it's not what I meant.
```suggestion
    def import_gds_3d(self, gds_file, gds_number, elevation=None, thickness=None, import_method=1):
```

File: pyaedt/application/Analysis3D.py (hunk 5 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1291]: done

File: pyaedt/application/Analysis3D.py (hunk 6 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1293]: Is it better like this;
        elevation_list : list, optional
            List of GDS layer elevation.
            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
            If elevation is none, elevation and thickness will not be applied.

File: pyaedt/application/Analysis3D.py (hunk 7 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1296]: thickness_list : list, optional
            List of GDS layer thickness.
            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
            If thickness is none, elevation and thickness will not be applied.

File: pyaedt/application/Analysis3D.py (hunk 8 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1322]: done

File: pyaedt/application/Analysis3D.py (hunk 9 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1372]: Done

File: pyaedt/application/Analysis3D.py (hunk 10 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1373]: How I can combine:

 for i in gds_number_list: 

and indices?

File: pyaedt/application/Analysis3D.py (hunk 3 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1281]: done

File: pyaedt/application/Analysis3D.py [hunk 11]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+
+        Parameters
+        ----------
+        gds_file : str
+            Path to the GDS file.
+        gds_number : list
+            List of layer numbers to import.
+        elevation : list, optional.
+            List of GDS layer elevation. The default is ``None``
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1292]:
```suggestion
            List of GDS layers elevation.
            Number of elements must be the same as gds_number, if not elevation and thickness will not be applied.
            The default is ``None``.
```

File: pyaedt/application/Analysis3D.py [hunk 12]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+            If elevation is None, elevation and thickness will not be applied.
+        thickness : list, optional
+            List of GDS layer thickness. The default is ``None``
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1295]:
```suggestion
            List of GDS layers thickness. 
            Number of elements must be the same as gds_number, if not elevation and thickness will not be applied.
            If thickness is ``None``, elevation and thickness will not be applied.
            The default is ``None``.
```

File: pyaedt/application/Analysis3D.py [hunk 13]
```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
...
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
+            If thickness is None, elevation and thickness will not be applied.
+        import_method : integer, optional
+            GDSII import method. The default is 1
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1299]:
```suggestion
            GDSII import method. 
            Options are: 
            - 0 for script
            - 1 for Parasolid
            
            The default is ``1``.
```

File: pyaedt/application/Analysis3D.py (hunk 10 above)
Comment (gmalinve) [reply pyaedt/application/Analysis3D.py:1373]: @amichel0205 gds_number is a list so you can access it through indeces: gds_number[j]

Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1378]: Giulia, can you please check the syntax is correct?

C:\AnsysDev\NoBackup\venvs\venv_pyaedt_dev\Scripts\python.exe C:\Users\amichel\AppData\Roaming\JetBrains\PyCharmCE2023.3\scratches\tes_nested_list.py 
  File "C:\Users\amichel\AppData\Roaming\JetBrains\PyCharmCE2023.3\scratches\tes_nested_list.py", line 19
    for(i=0;i<(len(gds_number));i++):
        ^^^
SyntaxError: invalid syntax. Maybe you meant '==' or ':=' instead of '='?

Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1378]: I found the solution: for:

 i in range(len(gds_number)):

File: pyaedt/application/Analysis3D.py (hunk 11 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1293]: done

File: pyaedt/application/Analysis3D.py (hunk 13 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py]: done

File: pyaedt/application/Analysis3D.py (hunk 12 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py]: done

File: pyaedt/application/Analysis3D.py (hunk 13 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py]: done

File: pyaedt/application/Analysis3D.py (hunk 10 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py]: done

File: pyaedt/application/Analysis3D.py [hunk 14]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+
+        Parameters
+        ----------
+        gds_file : str
+            Path to the GDS file.
+        gds_number : list
+            List of layer numbers to import.
+        elevation : list, optional.
+            List of GDS layer elevation.
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1292]: @amichel0205 elements

File: pyaedt/application/Analysis3D.py [hunk 15]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+        """Import a GDSII file.
+
+        Parameters
+        ----------
+        gds_file : str
+            Path to the GDS file.
+        gds_number : list
+            List of layer numbers to import.
+        elevation : list, optional.
+            List of GDS layer elevation.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1291]: @amichel0205 does it have to be plural elevation -> elevations? and same for thickness -> thicknesses?

File: pyaedt/application/Analysis3D.py [hunk 16]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
+            If thickness is "None", elevation and thickness will not be applied.
+            The default is ``None``
+        import_method : integer, optional
+            GDSII import method.
+            Options are:
+            - 0 = script
+            - 1 = Parasolid
+
+            The default is 1
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1306]: put 1 within `` and at end .

File: pyaedt/application/Analysis3D.py [hunk 17]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            The default is ``None``
+        thickness : list, optional
+            List of GDS layer thickness.
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
+            If thickness is "None", elevation and thickness will not be applied.
+            The default is ``None``
+        import_method : integer, optional
+            GDSII import method.
+            Options are:
+            - 0 = script
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1303]: put 0 within `` and 0 for script without the equal and same for next line

File: pyaedt/application/Analysis3D.py [hunk 18]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+        Parameters
+        ----------
+        gds_file : str
+            Path to the GDS file.
+        gds_number : list
+            List of layer numbers to import.
+        elevation : list, optional.
+            List of GDS layer elevation.
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
+            If elevation is "None", elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1293]: None within ``

File: pyaedt/application/Analysis3D.py [hunk 19]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            The default is ``None``
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1294]: . at the end

File: pyaedt/application/Analysis3D.py [hunk 20]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+        thickness : list, optional
+            List of GDS layer thickness.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1296]: @amichel0205 thicknesses?

File: pyaedt/application/Analysis3D.py [hunk 21]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            Number of element must be the same as gds_number list, if not elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1297]: @amichel0205 elements

File: pyaedt/application/Analysis3D.py [hunk 22]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            If thickness is "None", elevation and thickness will not be applied.
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1298]: use ``

File: pyaedt/application/Analysis3D.py [hunk 23]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+            The default is ``None``
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1299]: . at the end

File: pyaedt/application/Analysis3D.py [hunk 24]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+        import_method : integer, optional
+            GDSII import method.
+            Options are:
+            - 0 = script
+            - 1 = Parasolid
+
+            The default is 1
+
+                Returns
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1308]: intdentation

File: pyaedt/application/Analysis3D.py [hunk 25]
```diff
@@ -1277,6 +1277,141 @@ def import_dxf(
...
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+
+        if self.desktop_class.non_graphical:
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1314]:
```suggestion
        if  not self.desktop_class.non_graphical:
```

File: pyaedt/application/Analysis3D.py (hunk 17 above)
Comment (gmalinve) [reply pyaedt/application/Analysis3D.py:1303]: - ``0`` for script

File: pyaedt/application/Analysis3D.py (hunk 15 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1291]: done

File: pyaedt/application/Analysis3D.py (hunk 16 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1306]: done

File: pyaedt/application/Analysis3D.py (hunk 17 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1303]: done

File: pyaedt/application/Analysis3D.py (hunk 18 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1293]: done

File: pyaedt/application/Analysis3D.py (hunk 21 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1297]: done

File: pyaedt/application/Analysis3D.py (hunk 22 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1298]: done

File: pyaedt/application/Analysis3D.py (hunk 23 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1299]: done

File: pyaedt/application/Analysis3D.py (hunk 19 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1294]: done

File: pyaedt/application/Analysis3D.py (hunk 24 above)
Comment (amichel0205) [reply pyaedt/application/Analysis3D.py:1308]: done

**Review by Samuelopez-ansys — COMMENTED** [review]
@amichel0205 

Why rather than having 3 lists you don't use 2 dictionaries?

**Comment by amichel0205** [comment]
> @amichel0205
> 
> Why rather than having 3 lists you don't use 2 dictionaries?

@Samuelopez-ansys @gmalinve 

Giulia told me the same 😊
Next time.
I've to practice.

**Comment by amichel0205** [comment]
closed by mistake 😒

File: pyaedt/application/Analysis3D.py [hunk 26]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oEditor.oeditor.ImportGDSII
```
Comment (gmalinve) [inline pyaedt/application/Analysis3D.py:1316]: oEditor.ImportGDSII

**Comment by amichel0205** [comment]
> @amichel0205 Good that we are good reviewers :) You have the opportunity to improve your pull request. Let us know if you have any question. We can not merge your pull request until the modifications are performed.
> 
> Thanks for your contribution!

Ok, you are pushing me hard but I will do it. 😉

File: _unittest/test_20_HFSS.py [hunk 27]
```diff
@@ -1624,3 +1624,15 @@ def test_67_transient_composite(self, add_app):
...
+
+    @pytest.mark.skipif(config["NonGraphical"], reason="Test fails on build machine")
+    def test_68_import_gds_3d(self):
+        self.aedtapp.insert_design("gds_import_H3D")
+        gds_file = os.path.join(local_path, "example_models", "cad", "GDS", "gds1.gds")
+        assert self.aedtapp.import_gds_3d(gds_file, [7, 9], [100e-6, 110e-6], [10e-6, 5e-6], 0)
+        assert self.aedtapp.import_gds_3d(gds_file, [7, 9])
+        assert self.aedtapp.import_gds_3d(gds_file, [7], [100e-6, 110e-6], [10e-6, 5e-6], 1)
+        gds_number = []
+        assert not self.aedtapp.import_gds_3d(gds_file, [])
```
Comment (MaxJPRey) [inline _unittest/test_20_HFSS.py:1636]:
```suggestion
        assert not self.aedtapp.import_gds_3d(gds_file, [])
```

File: pyaedt/application/Analysis3D.py [hunk 28]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+            Number of elements must be the same as gds_number list, if not elevation and thickness will not be applied.
+            If thickness is ``None``, elevation and thickness will not be applied.
+            The default is ``None``.
+        import_method : integer, optional
+            GDSII import method.
+            Options are:
+            - ``0`` for script
+            - ``1`` for Parasolid
+
+            The default is ``1``
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1306]:
```suggestion
        elevation : list, optional
            List of GDS layer elevations. The number of elements in the list must be
            the same as the number of elements in the list for the ``gds_number``
            parameter.  If the number is not the same, elevation and thickness are
            not applied. If ``elevation=None``, which is the default, elevation and
            thickness are not applied.
        thickness : list, optional
            List of GDS layer thicknesses. The number of elements in the list must
            be the same as the number of elements in the list for the ``gds_number``
            parameter. If the number is not the same, elevation and thickness are not
            applied. If ``thickness=None``, which is the default, elevation and thickness
            are not applied.
        import_method : integer, optional
            GDSII import method. The default is ``1``. Options are:
            
            - ``0`` for script
            - ``1`` for Parasolid
                      
```
Avoid future tense. For a list to render correctly in the doc, a blank line must precede the first list item and follow the last list item.

File: pyaedt/application/Analysis3D.py [hunk 29]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+        >>> oEditor.ImportGDSII
+
+        """
+
+        if self.desktop_class.non_graphical:
+            self.logger.error("Method is supported only in graphical mode.")
+            return False
+
+        if not os.path.exists(gds_file):
+            self.logger.error("GDSII file does not exist, no layer imported ")
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1325]:
```suggestion
            self.logger.error("GDSII file does not exist. No layer is imported.")
```

File: pyaedt/application/Analysis3D.py [hunk 30]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+            return False
+        if len(gds_number) == 0:
+            self.logger.error("GDSII layer number list is empty, no layer imported ")
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1328]:
```suggestion
            self.logger.error("List for GDSII layer numbers is empty. No layer is imported.")
```

File: pyaedt/application/Analysis3D.py [hunk 31]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+
+        if (
+            elevation is None
+            or thickness is None
+            or (len(gds_number) != len(elevation) or len(gds_number) != len(thickness))
+        ):
+            self.logger.warning(
+                "One of parameter list: elevation and/or thickness is either: "
+                "not specified or has different number of elements than GDS number."
+                "Elevation and thickness will not be applied "
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1340]:
```suggestion
                "A list of elevations and/or thicknesses is either "
                "not specified or has a different number of elements than the GDS number."
                "Elevation and thickness are applied."
```

File: pyaedt/application/Analysis3D.py [hunk 32]
```diff
@@ -1277,6 +1277,147 @@ def import_dxf(
...
+                    gds_file,
+                    "FlattenHierarchy:=",
+                    True,
+                    "ImportMethod:=",
+                    import_method,
+                    layermap,
+                ]
+            )
+        else:
+            self.logger.info("GDS layer imported with elevation and thickness")
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1370]:
```suggestion
            self.logger.info("GDS layer imported with elevation and thickness.")
```

**Review by Samuelopez-ansys — DISMISSED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]
